src/Lecture4.hs

```diff
@@ -237,7 +313,16 @@ you can return string "no value"
 -}
 
 displayStats :: Stats -> String
-displayStats = error "TODO"
+displayStats s =
+      "Total positions:       : " ++ show (getSum (statsTotalPositions s)) ++ "\n"
+   ++ "Total final balance    : " ++ show (getSum (statsTotalSum s))       ++ "\n"
+   ++ "Biggest absolute cost  : " ++ show (getMax (statsAbsoluteMax s))    ++ "\n"
+   ++ "Smallest absolute cost : " ++ show (getMin (statsAbsoluteMin s))    ++ "\n"
+   ++ "Max earning            : " ++ if isNothing (statsSellMax s) then "no value" else show (getMax (fromJust (statsSellMax s))) ++ "\n"
```

Comment:
`isNothing` + `fromJust` is considered a "boolean blindness" anti-patterns. You can use standard function `maybe` here instead:

```suggestion
   ++ "Max earning            : " ++ maybe "no value" (show . getMax) (statsSellMax s) ++ "\n"
```